client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java

```diff
@@ -96,7 +97,7 @@ public void tokenReceived(URI tokenUri)
                     .withMaxAttempts(-1)
                     .withMaxDuration(Duration.ofSeconds(4))
                     .withBackoff(100, 500, MILLIS)
-                    .handleResultIf(code -> code != HTTP_OK))
+                    .handleResultIf(code -> code >= HTTP_INTERNAL_ERROR))
```

Comment:
what about 3xx and 4xx?

Reply:
it should always be a kind of fire and forget, so there is no point in retrying for 300 or 400 as we won't handle that either way.

Reply:
Exactly. There is no point in retrying 300 and 400s